src/app/lib/views/browser/item.js

```diff
@@ -77,247 +71,216 @@
             }
 
+            // display the spinner
+            $('.spinner').show();
+            provider.detail(this.model.get('imdb_id'), this.model.attributes).then(function (data) {
+                $('.spinner').hide();
+
+                // inject provider's name
+                data.provider = provider.name;
+
+                // load details
+                App.vent.trigger(itemtype + ':showDetail', new App.Model[modelType](data));
+            }).catch(function (err) {
+                win.error(err);
+                $('.spinner').hide();
+                $('.notification_alert').text(i18n.__('Error loading data, try again later...')).fadeIn('fast').delay(2500).fadeOut('fast');
+            });
         },
 
         toggleWatched: function (e) {
             e.stopPropagation();
             e.preventDefault();
-            var that = this;
-            if (this.model.get('watched')) {
+
+            var watched = !this.model.get('watched');
+            var imdb = this.model.get('imdb_id');
+            var dbCall = watched ? 'markMovieAsWatched' : 'markMovieAsNotWatched';
+            var appEvent = watched ? 'movie:watched' : 'movie:unwatched';
+
+            // add/delete db item
+            Database[dbCall]({
+                imdb_id: imdb,
+                from_browser: true
+            }, true).then(function () {
+                // set watched state
+                this.model.set('watched', watched);
+
+                // reset cover state to default
                 this.ui.watchedIcon.removeClass('selected');
-                if (Settings.watchedCovers === 'fade') {
-                    this.$el.removeClass('watched');
-                }
-                Database.markMovieAsNotWatched({
-                        imdb_id: this.model.get('imdb_id')
-                    }, true)
-                    .then(function () {
-                        that.model.set('watched', false);
-                        App.vent.trigger('movie:unwatched', {
-                            imdb_id: that.model.get('imdb_id')
-                        }, 'seen');
+                this.$el.removeClass('watched');
+
+                // set cover state
+                this.setCoverStates();
+                this.setTooltips();
+
+                // event propagation
+                App.vent.trigger(appEvent, {
+                    imdb_id: imdb
+                }, 'seen');
+            }.bind(this));
+        },
+
+        addBookmarked: function () {
+            var imdb = this.model.get('imdb_id');
+            var itemtype = this.model.get('type');
+            var provider = App.Providers.get(this.model.get('provider'));
+
+            return provider.detail(imdb, this.model.attributes).then(function (data) {
+                data.provider = provider.name;
+
+                if (itemtype === 'show') {
+                    promisifyDb(db.shows.find({
+                        imdb_id: imdb.toString(),
```

Comment:
i'd rather have no direct calls to db here, let's move that into db and call a helper function from here ?